tests/gold_tests/thread_config/check_threads.py

```diff
@@ -0,0 +1,83 @@
+'''
+'''
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import psutil
+import argparse
+
+
+def count_threads(ts_path, etnet_threads, accept_threads):
+
+    for pid in psutil.pids():
+
+        # Find the pid corresponding to the ats process we started in autest.
+        # It needs to match the process name and the binary path.
+        # If autest can expose the pid of the process this is not needed anymore.
+        p = psutil.Process(pid)
+        if p.name() == '[TS_MAIN]' and p.cwd() == ts_path:
+
+            etnet_check = set()
+            accept_check = set()
+
+            for t in p.threads():
+
+                # Get the name of the thread.
+                thread_name = psutil.Process(t.id).name()
+
+                if thread_name.startswith('[ET_NET'):
+
+                    # Get the id of this thread and check if it's in range.
+                    etnet_id = int(thread_name.split(' ')[1][:-1])
+                    if etnet_id >= etnet_threads:
+                        return 2
```

Comment:
It might be beneficial to write a simple message to `stderr` describing why we're exiting in each case, so that the dev who is debugging can see at a glance why this check did not pass.